Realm/RealmSwift/Migration.swift

```diff
@@ -18,31 +18,111 @@
 
 import Realm
 
+/// A block that is used to migrate objects from one version of a Realm to a newer version.
 public typealias MigrationBlock = (migration: Migration, oldSchemaVersion: UInt) -> Void
 
+/**
+Specify a schema version and an associated migration block which is applied when
+opening any Realm with an old schema version.
+
+Before you can open an existing `Realm` which has a different on-disk schema
+from the schema defined in your object interfaces you must provide a migration
+block which converts from the disk schema to your current object schema. At the
+minimum your migration block must initialize any properties which were added to
+existing objects without defaults and ensure uniqueness if a primary key
+property is added to an existing object.
+
+You should call this method before accessing any `Realm` instances which
+require migration. After registering your migration block Realm will call your
+block automatically as needed.
+
+:warning: Unsuccessful migrations will throw exceptions when the migration block
+          is applied. This will happen in the following cases:
+- The migration block was run and returns a schema version which is not higher
+  than the previous schema version.
+- A new property without a default was added to an object and not initialized
+  during the migration. You are required to either supply a default value or to
+  manually populate added properties during a migration.
+
+:param: schemaVersion  The current schema version.
+:param: migrationBlock The block which migrates the Realm to the current version.
+*/
 public func setSchemaVersion(schemaVersion: UInt, migrationBlock: MigrationBlock) {
     RLMRealm.setSchemaVersion(schemaVersion, withMigrationBlock: {
         migrationBlock(migration: Migration($0), oldSchemaVersion: $1)
     })
 }
 
+/**
+Performs the registered migration block on the Realm at the given path.
+
+This method is called automatically when opening a Realm for the first time and does
+not need to be called explicitly. You can choose to call this method to control
+exactly when and how migrations are performed.
+
+:see: setSchemaVersion(_:_:)
+
+:param: realmPath The path of the Realm to migrate.
+
+:returns: The error that occurred while applying the migration if any.
+*/
 public func migrateRealm(path: String) -> NSError? {
     return RLMRealm.migrateRealmAtPath(path)
 }
 
+/**
+Migration encapsulates the information passed into a MigrationBlock when updating the schema version of a Realm instance.
+*/
 public class Migration {
+    // MARK: Properties
+
+    /// The old schema.
+    /// This is the schema that describes the Realm before the migration is applied.
```

Comment:
Remove the first line here. Total comment could just read `The schema that describes the Realm before the migration is applied.`. Same for the new schema one.